cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/SinkEmitter.java

```diff
@@ -52,18 +48,12 @@ public void emit(Object value) {
 
   @Override
   public void emitError(InvalidEntry invalidEntry) {
-   // Not supported
+    // Not supported - This should never happen
     LOG.error("Emitting errors from sink {} is not supported", stageName);
   }
 
   @Override
-  public void addTransformDetail(String stageName, BatchTransformDetail batchTransformDetail) {
-    //no-op
-  }
-
-  @Nullable
-  @Override
-  public Map<String, BatchTransformDetail> getNextStages() {
-    return null;
+  public void addTransformDetail(String stageName, PipeTransformDetail pipeTransformDetail) {
+    throw new IllegalStateException(String.format("Sink emitter does not have {} as output", stageName));
```

Comment:
nit: "Sink {} should not have any output, but it has output {}", this.stageName, stageName